tests-pending/rpc_pending_test.go

```diff
@@ -0,0 +1,332 @@
+	param[0]["to"] = addrA
+	param[0]["value"] = "0xA"
+	param[0]["gasLimit"] = "0x5208"
+	param[0]["gasPrice"] = "0x1"
+
+	_ = util.Call(t, "eth_sendTransaction", param)
+
+	rpcRes := util.Call(t, "eth_getTransactionByBlockNumberAndIndex", []interface{}{"pending", "0x1"})
+	var pendingBlock map[string]interface{}
+	err := json.Unmarshal(rpcRes.Result, &pendingBlock)
+	require.NoError(t, err)
+
+	require.Equal(t, pendingBlock["blockHash"], nil)
+	require.Equal(t, pendingBlock["blockNumber"], nil)
+	require.Equal(t, pendingBlock["transactionIndex"], nil)
+	require.NotEmpty(t, pendingBlock["hash"])
+
+	rpcRes = util.Call(t, "eth_getTransactionByBlockNumberAndIndex", []interface{}{"latest", "0x1"})
+	var latestBlock map[string]interface{}
+	err = json.Unmarshal(rpcRes.Result, &latestBlock)
+	require.NoError(t, err)
+
+	require.Empty(t, latestBlock)
+}
+
+func TestEth_Pending_GetTransactionByHash(t *testing.T) {
+	param := make([]map[string]string, 1)
+	param[0] = make(map[string]string)
+	param[0]["from"] = "0x" + fmt.Sprintf("%x", from)
+	param[0]["to"] = addrA
+	param[0]["value"] = "0xA"
+	param[0]["gasLimit"] = "0x5208"
+	param[0]["gasPrice"] = "0x1"
+
+	txRes := util.Call(t, "eth_sendTransaction", param)
+	var txHash common.Hash
+	err := txHash.UnmarshalJSON(txRes.Result)
+	require.NoError(t, err)
+
+	rpcRes := util.Call(t, "eth_getTransactionByHash", []interface{}{txHash})
+	var pendingBlock map[string]interface{}
+	err = json.Unmarshal(rpcRes.Result, &pendingBlock)
+	require.NoError(t, err)
+
+	require.NotEmpty(t, pendingBlock)
+	require.Equal(t, nil, pendingBlock["blockHash"])
+	require.Equal(t, nil, pendingBlock["blockNumber"])
+	require.Equal(t, nil, pendingBlock["transactionIndex"])
+	require.NotEmpty(t, pendingBlock["hash"])
+	require.NotEmpty(t, pendingBlock["value"], "0xa")
+}
+
+func TestEth_Pending_SendTransaction_PendingNonce(t *testing.T) {
+	currNonce := util.GetNonce(t, "latest")
+	param := make([]map[string]string, 1)
+	param[0] = make(map[string]string)
+	param[0]["from"] = "0x" + fmt.Sprintf("%x", from)
+	param[0]["to"] = addrA
+	param[0]["value"] = "0xA"
+	param[0]["gasLimit"] = "0x5208"
+	param[0]["gasPrice"] = "0x1"
+
+	// first transaction
+	_ = util.Call(t, "eth_sendTransaction", param)
+	pendingNonce1 := util.GetNonce(t, "pending")
+	require.Greater(t, uint64(pendingNonce1), uint64(currNonce))
+
+	// second transaction
+	param[0]["to"] = "0x7f0f463c4d57b1bd3e3b79051e6c5ab703e803d9"
+	_ = util.Call(t, "eth_sendTransaction", param)
+	pendingNonce2 := util.GetNonce(t, "pending")
+	require.Greater(t, uint64(pendingNonce2), uint64(currNonce))
+	require.Greater(t, uint64(pendingNonce2), uint64(pendingNonce1))
+
+	// third transaction
+	param[0]["to"] = "0x7fb24493808b3f10527e3e0870afeb8a953052d2"
+	_ = util.Call(t, "eth_sendTransaction", param)
+	pendingNonce3 := util.GetNonce(t, "pending")
+	require.Greater(t, uint64(pendingNonce3), uint64(currNonce))
+	require.Greater(t, uint64(pendingNonce3), uint64(pendingNonce2))
```

Comment:
maybe do checks for a hard value instead of just greater, same with above

Reply:
the problem with hard values is that there may be pending transactions leftover from previous tests... so unless there's a convenient way to flush the pending transaction queue or wait 150 sec for each test. I try to complete all the test within the same block because if a new block is produced in between, the test will fail